src/SearchView.js

```diff
@@ -0,0 +1,127 @@
+import Gtk from "gi://Gtk";
+import GObject from "gi://GObject";
+
+import { hasMatch, score } from "./fzy.js";
+
+import Template from "./SearchView.blp" with { type: "uri" };
+
+const QUERY_TYPES = [
+  "additional",
+  "alias",
+  "bitfield",
+  "callback",
+  "class",
+  "constant",
+  "constructor",
+  "enum",
+  "error",
+  "function",
+  "interface",
+  "namespace",
+  "macro",
+  "method",
+  "property",
+  "signal",
+  "struct",
+  "union",
+  "vfunc",
+];
+
+const QUERY_PATTERN = new RegExp(
+  "^(" + QUERY_TYPES.join("|") + ")\\s*:\\s*",
+  "i",
+);
+
+export const SearchView = GObject.registerClass(
+  {
+    GTypeName: "SearchView",
+    Template,
+    Properties: {
+      "search-term": GObject.ParamSpec.string(
+        "search-term",
+        "search-term",
+        "Current active search term",
+        GObject.ParamFlags.READWRITE,
+        "",
+      ),
+    },
+    Signals: {
+      "search-view-selection-changed": {
```

Comment:
Similarly, I also listen for this signal in ``Sidebar.js`` to select an item in the ``BrowseView`` when something gets selected in the ``SearchView``, so ``Sidebar.js`` acts as a centralized place for communication for both the views